tests/testthat/test-fims-estimation.R

```diff
@@ -261,31 +261,29 @@ test_that("deterministic test of fims", {
   }
 
   # Expected survey index
-  fims_object <- report$expected_index[, 2]
-
-  cwaa <- matrix(report$cwaa[1:(om_input$nyr * om_input$nages), 2], nrow = om_input$nyr, byrow = T)
-  expect_equal(fims_object, apply(cwaa, 1, sum) * om_output$survey_q$survey1)
+  cwaa <- matrix(report$cwaa[[2]][1:(om_input$nyr * om_input$nages)], nrow = om_input$nyr, byrow = T)
+  expect_equal(fims_index[[2]], apply(cwaa, 1, sum) * om_output$survey_q$survey1)
```

Comment:
I find the nested vectors from different fleets a bit confusing. Should we consider adding an additional layer of nesting for fleets? This way, `fims_index[[1]][[1]]` and `fims_index[[1]][[2]]` would correspond to indices 1 and 2 from population 1, related to `biomass[[1]]`, while `fims_index[[2]][[1]]` and `fims_index[[2]][[2]]` would correspond to indices 1 and 2 from population 2, related to `biomass[[2]]`. With the current setup, are we planning to continue expanding the fims_index IDs to accommodate more fleets for different populations (e.g., `fims_index[[3]]` and `fims_index[[4]]` for population 2)?

Reply:
In the current set-up, fims_index[[1]] corresponds to the fleet module with an id == 1 (fishing_fleet in test_fims_estimation.R), that is:

`
fims_index[[1]] == fims_index[[fishing_fleet$get_id()]]
fims_index[[2]] == fims_index[[survey_fleet$get_id()]]
`

I was thinking it would be easier to keep track on the R side if we keep this 1:1 mapping of outputs and modules. We should probably figure out how we are going to connect populations and fleets when there are multiple populations, each with multiple, independent fleets.

The other way of reporting out derived variables from fleet is to report a matrix where the number of columns correspond to the number of fleets. The reason for using the ragged array approach (using nested vectors) is that it would allow the dimensions (e.g. numbers of years) to vary by fleet. The matrix approach will still work. If the dimensions do not match, the matrix dimensions can be set based on the largest dimension and missing values filled in with -999. These are treated as constants and dropped out of standard error calculations. 

I can change the code to use a matrix approach if this is easier to understand. I think an expansion that accounts for multiple populations, each with multiple fleets can be handled in a separate issue.

Reply:
Thank you for the clarification! I think the ragged array approach works for the current version of FIMS. I agree that addressing the expansion to accommodate multiple populations, each with multiple fleets, can be a separate issue.